src/core/abr/representation_chooser.ts

```diff
@@ -433,107 +254,103 @@ export default class RepresentationChooser {
             representations[0],
           manual: true,
           urgent: true, // a manual bitrate switch should happen immediately
+          lastStableBitrate: Infinity,
         });
       }
 
       // -- AUTO mode --
-      let inStarvationMode = false; // == buffer gap too low == panic mode
+      let lastEstimatedBitrate: number|undefined;
+      let lastEstimationMode: "bandwidth"|"BOLA"|undefined = "bandwidth";
+      const bufferBasedEstimation$ =
+        BufferBasedChooser(
+          appendSegment$,
+          this._scoreData$.pipe(filter((e): e is {
+            representation: Representation;
+            EWMA: EWMA;
+          } => !!e)),
+          representations
+        );
+
       return observableCombineLatest(
         clock$,
         maxAutoBitrate$,
         deviceEvents$,
-        this._reEstimate$.pipe(startWith(null))
+        bufferBasedEstimation$.pipe(startWith(undefined)),
+        this._scoreData$.pipe(startWith(undefined))
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this redundant "undefined". ([typescript:S4623](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4623))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=442&issues=AWg4IEcySi5GWQS9ds0x&open=AWg4IEcySi5GWQS9ds0x)

[ISSUE_KEY]: AWg4IEcySi5GWQS9ds0x